ERCS/erc-7844.md

```diff
@@ -0,0 +1,529 @@
+
+
+## Specification 
+
+![Adaptability](../assets/erc-7844/Adaptability.svg)
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY” and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Extendable Structs and Storage Spaces
+- Extendable structs **MUST** leverage dynamic mappings with deterministic field hashes (`keccak256`).  
+- The base struct and its additions **MUST** remain immutable, while fields **MUST** be dynamically appendable.
+- Storage spaces **MUST** equate to a simple extension of the hashing structure, segmenting both struct-defining and active data.
+
+Each struct member **MUST** be defined using the following compact metadata structures:
+
+| **Field**  | **Bits** | **Description**                          |
+| ---------- | -------- | ---------------------------------------- |
+| `bitCount` | 128      | Starting bit offset for the member.      |
+| `size`     | 64       | Size of the member in bits.              |
+| `type`     | 64       | Type ID (e.g., `uint256`, `bool`, etc.). |
+
+### Type IDs:
+
+| **Type**                   | **ID** | **Size**     |
+| -------------------------- | ------ | ------------ |
+| `uint`                     | 1      | 8...256 bits |
+| `int`                      | 2      | 8...256 bits |
+| `bool`                     | 3      | 8 bits       |
+| `address`                  | 4      | 160 bits     |
+| `bytes32 (optional)`       | 5      | 256 bits     |
+| `string (≡ bytes)`         | 6      | Dynamic      |
+
+*For arrays, developers **MAY** define unpacking logic to treat `string` or `bytes` fields as indexed collections of dynamic elements.*
+
+### Hash Structure:
+There are three main segments that **MUST** include separate hashing structures to resolve any potential for collisions, meta-segmented by their particular storage spaces.  These are: storage space state data and member-specific data, storage space live data, and storage space dynamic data. 
+
+There are two special values that **MUST** be included: a `safeIndex` and a `stringIndex`.  Both are expounded upon in detail below.
+
+Developers **SHOULD** include a unique hash offset for each space, as this simplifies the development and audit process considerably, and thereby greatly reduces the risk of improper implementation.
+
+A `storageSpace` `offset` value **MUST** be included.  This is explicitly marked in the below section for clarity.  
+
+*MEMBERS_LIM* refers to the total max quantity of struct members permitted within a single extendible struct.  
+*ENTRIES_LIM* refers to the max amount of entries in a storage space, and serves as the upper limit for *push* and *pushMany*.  
+
+`A`, `B`, `C` hash offset structure not only allows CDS to functionally guarantee safety from collisions, but also aid in debugging by 
+allowing maintainers to instantly track the high-level objective of complex storage operations.
+
+**Metadata Root:**  
```

Comment:
Re: introductory text, you definitely need some here. You're jumping straight into Yul.